frontend/src/components/pod/List.tsx

```diff
@@ -111,3 +118,135 @@ export default function PodList() {
 
   return <PodListRenderer pods={pods} error={error} />;
 }
+
+interface PodTableRow {
+  restartsStr: string;
+  reason: string;
+  totalContainers: number;
+  readyContainers: number;
+  lastRestartDate: Date;
+}
+
+// Source Code: https://github.com/kubernetes/kubernetes/blob/33bf8a8ebd33eae0d45c18feb1c05baa6a055cde/pkg/printers/internalversion/printers.go#L558
+function getPodTableInfo(pod: KubePod): PodTableRow {
+  let restarts = 0;
+  const totalContainers = (pod.spec.containers ?? []).length;
+  let readyContainers = 0;
+  let lastRestartDate = new Date(0);
+
+  let reason = pod.status.phase;
+  if (!!pod.status.reason) {
+    reason = pod.status.reason;
+  }
```

Comment:
This can be `let reason = pod.status.reason ? pod.status.reason :  pod.status.phase;`